crates/metal-mir/src/structure.rs

```diff
@@ -0,0 +1,38 @@
+use crate::types::{visibility::Visibility, Type};
+
+/// Represents the properties of a struct.
+/// i.e. `a: B`
+#[derive(Debug, Clone)]
+pub struct StructProperty {
+    /// name of the struct property. i.e.: `a`
+    pub name: &'static str,
+    /// index of the struct property (order,) i.e. `0`
+    pub idx: u32,
```

Comment:
I don't think this is used anywhere?

Reply:
This is a left over from my attempts at implementing enums. Will remove for now, however.